File: Java/src/main/java/TestApplitoolsWebsite.java
import com.applitools.eyes.Eyes;
import com.applitools.eyes.RectangleSize;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;

import java.net.URISyntaxException;

public class TestApplitoolsWebsite {

    public static void main(String[] args) throws URISyntaxException, InterruptedException {

        WebDriver driver = new FirefoxDriver();

        Eyes eyes = new Eyes();
        // This is your api key, make sure you use it in all your tests.
        eyes.setApiKey(APPLITOOLS_APIKEY);

        try {
            // Start visual testing with browser viewport set to 1024x768.
            // Make sure to use the returned driver from this point on.
            driver = eyes.open(driver, "Applitools", "Test Web Page", new RectangleSize(900, 600));

            driver.get("http://applitools.com");

            // Visual validation point #1
            eyes.checkWindow("Main Page");

            driver.findElement(By.cssSelector(".read_more")).click();

            // Visual validation point #2
            eyes.checkWindow("Features page");

            // End visual testing. Validate visual correctness.
            eyes.close();
        } finally {
            // Abort test in case of an unexpected error.
            eyes.abortIfNotClosed();
            driver.quit();
        }
    }
}

File: Ruby/app/test_applitools_website.rb
require 'eyes_selenium'

eyes = Applitools::Eyes.new
# This is your api key, make sure you use it in all your tests.
eyes.api_key = APPLITOOLS_APIKEY

# Get a selenium web driver object.
my_webdriver = Selenium::WebDriver.for :firefox

#baseline = os + browser + viewport-size + app_name + test
begin
  # Start visual testing using my_webdriver and setting the viewport to 1024x768.
  eyes.test(app_name: 'Applitools website', test_name: 'Example test',
            viewport_size: {width: 900, height: 600}, driver: my_webdriver) do |driver|

    driver.get 'http://www.applitools.com'
    # Visual validation point #1
    eyes.check_window('Main Page')
    driver.find_element(:css, ".read_more").click
    # Visual validation point #2
    eyes.check_window('Features Page')
  end
ensure
  eyes.abort_if_not_closed
  my_webdriver.quit
end
